internal/it_automation/it_automation_policy_resource.go

```diff
@@ -0,0 +1,891 @@
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+// updatePolicyEnabledState enables or disables a policy.
+func (r *itAutomationPolicyResource) updatePolicyEnabledState(
+	ctx context.Context,
+	policyID string,
+	enabled bool,
+) error {
+	action := "enabling"
+	if !enabled {
+		action = "disabling"
+	}
+
+	tflog.Info(
+		ctx,
+		fmt.Sprintf(
+			"Starting %s operation for policy ID: %s",
+			action,
+			policyID,
+		),
+	)
+
+	body := &models.ItautomationUpdatePolicyRequest{
+		ID:        policyID,
+		IsEnabled: &enabled,
+	}
+
+	params := &it_automation.ITAutomationUpdatePoliciesParams{
+		Context: ctx,
+		Body:    body,
+	}
+
+	_, err := r.client.ItAutomation.ITAutomationUpdatePolicies(params)
+	if err != nil {
+		return fmt.Errorf("error %s policy ID %s: %w", action, policyID, err)
+	}
+
+	tflog.Info(
+		ctx,
+		fmt.Sprintf(
+			"Successfully %s policy %s",
+			action,
+			policyID,
+		),
+	)
+
+	return nil
+}
+
+// Delete deletes the resource and removes the Terraform state on success.
+func (r *itAutomationPolicyResource) Delete(
+	ctx context.Context,
+	req resource.DeleteRequest,
+	resp *resource.DeleteResponse,
+) {
+	var state itAutomationPolicyResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	policyID := state.ID.ValueString()
+
+	currentPolicy, diags := getItAutomationPolicy(ctx, r.client, policyID)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if currentPolicy.IsEnabled {
+		err := r.updatePolicyEnabledState(ctx, policyID, false)
+		if err != nil {
+			if isNotFoundError(err) {
+				tflog.Warn(
+					ctx,
+					fmt.Sprintf("%s %s not found during disable, removing from state", itAutomationPolicy, policyID),
+					map[string]any{"error": err.Error()},
+				)
+				resp.State.RemoveResource(ctx)
+				return
+			}
+
+			resp.Diagnostics.AddError(
+				"Error disabling IT automation policy before deletion",
+				err.Error(),
+			)
+			return
+		}
+
+		verifyPolicy, diags := getItAutomationPolicy(ctx, r.client, policyID)
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+
+		if verifyPolicy.IsEnabled {
+			resp.Diagnostics.AddError(
+				"Policy disable verification failed",
+				fmt.Sprintf(
+					"Policy %s still shows IsEnabled=true after disable operation",
+					policyID,
+				),
+			)
+			return
+		}
+	}
+
+	params := &it_automation.ITAutomationDeletePolicyParams{
+		Context: ctx,
+		Ids:     []string{policyID},
+	}
+
+	_, err := r.client.ItAutomation.ITAutomationDeletePolicy(params)
+	if err != nil {
+		if strings.Contains(strings.ToLower(err.Error()), "not found") ||
+			strings.Contains(err.Error(), "404") {
+			tflog.Warn(
+				ctx,
+				fmt.Sprintf(notFoundRemoving, fmt.Sprintf("%s %s", itAutomationPolicy, policyID)),
+				map[string]any{"error": err.Error()},
+			)
+			resp.State.RemoveResource(ctx)
+			return
+		}
```

Comment:
we don't have to remove the resource from state on delete. Terraform core handles this. 
```suggestion
		if strings.Contains(strings.ToLower(err.Error()), "not found") ||
			strings.Contains(err.Error(), "404") {
			return
		}
```